compiler/modules/front/resolveTypes.bal

```diff
@@ -148,8 +148,36 @@ function resolveTypeDesc(ModuleSymbols mod, s:ModuleLevelDefn modDefn, int depth
             t:ListDefinition d = new;
             td.defn = d;
             t:SemType[] members = from var x in td.members select check resolveTypeDesc(mod, modDefn, depth + 1, x);
-            t:SemType rest = check resolveTypeDesc(mod, modDefn, depth + 1, td.rest);
-            return d.define(env, members, rest);
+            t:SemType rest = t:NEVER;
+            s:TypeDesc? restTd = td.rest;
+            if restTd != () {
+                rest = check resolveTypeDesc(mod, modDefn, depth + 1, restTd);
+            }
+            return d.define(env, initial = members, rest = rest);
+        }
+        else {
+            return defn.getSemType(env);
+        }   
+    }    
+    if td is s:ArrayTypeDesc {
+        t:ListDefinition? defn = td.defn;
+        if defn == () {
+            t:ListDefinition d = new;
+            td.defn = d;
+            t:SemType t =  check resolveTypeDesc(mod, modDefn, depth + 1, td.member);
+            foreach s:SimpleConstExpr? len in td.dimensions.reverse() {
+                if len == () {
+                    t = d.define(env, rest = t);
+                }
+                else {
+                    if !mod.allowAllTypes {
+                        return err:unimplemented("fixed length array types not implemented", s:locationInDefn(modDefn, td.startPos));
+                    }
+                    [t:SemType, t:Value] [_, resolved] = check resolveConstExpr(mod, modDefn, len, t:INT);
```

Comment:
Localize the cast by adding to const.bal a function resolveConstIntExpr that returns int|FoldError.